crates/astria-composer/src/searcher/executor.rs

```diff
@@ -164,28 +205,43 @@ impl Executor {
                             break Err(e).wrap_err("failed submitting bundle to sequencer");
                         }
                     }
+
+                    block_timer.as_mut().reset(Instant::now() + Duration::from_millis(self.block_time_ms));
+                }
+
+                Some(next_bundle) = future::ready(bundle_factory.next_finished()), if submission_fut.is_terminated() => {
+                    let bundle = next_bundle.pop();
+                    if !bundle.is_empty() {
+                        submission_fut = self.submit_bundle(nonce, bundle);
+                    }
+                }
+
+                // receive new seq_action and bundle it
+                Some(seq_action) = self.serialized_rollup_transactions_rx.recv() => {
+                    let rollup_id = seq_action.rollup_id;
+                    match bundle_factory.try_push(seq_action) {
+                        Ok(()) => {}
+                        Err(BundleFactoryError::SequenceActionTooLarge{size, max_size}) => {
+                            warn!(
+                                rollup_id = %rollup_id,
+                                seq_action_size = size,
+                                max_size = max_size,
+                                "failed to bundle sequence action: too large. sequence action is dropped."
+                            );
+                        }
+                    }
                 }
 
-                // receive new bundle for processing
-                Some(bundle) = self.new_bundles.recv(), if submission_fut.is_terminated() => {
-                    // TODO(https://github.com/astriaorg/astria/issues/476): Attach the hash of the
-                    // bundle to the span. Linked GH issue is for agreeing on a hash for `SignedTransaction`,
-                    // but both should be addressed.
-                    let span =  info_span!(
-                        "submit bundle",
-                        nonce.initial = nonce,
-                        bundle.len = bundle.len(),
-                    );
-                    submission_fut = SubmitFut {
-                        client: self.sequencer_client.clone(),
-                        address: self.address,
-                        nonce,
-                        signing_key: self.sequencer_key.clone(),
-                        state: SubmitState::NotStarted,
-                        bundle,
+                // try to preempt current bundle if the timer has ticked without submitting the next bundle
+                () = &mut block_timer, if submission_fut.is_terminated() => {
+                    let bundle = bundle_factory.pop_now();
+                    if !bundle.is_empty() {
```

Comment:
Just like above, I think  it might be valuable info to notify that the timer was up, but the bundle was empty